src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -149,19 +151,19 @@ func (a *StorageMinerActorCode_I) _onMissedSurprisePoSt(rt Runtime) {
 
 	Release(rt, h, st)
 
+	a._submitPowerReport(rt, lastPoStResponse)
+
 	// Note: NewDetectedFaults is now the sum of all
 	// previously active, committed, and recovering sectors minus expired ones
 	// and any previously Failing sectors that did not exceed MaxFaultCount
 	// Note: previously declared faults is now treated as part of detected faults
```

Comment:
Aside: I would really appreciate a write-up of the fault and slashing rules that could surface things like this more clearly. I just noticed this one and think we should reconsider.

Reply:
+1, reworking md would be dope